includes/class-freemius.php

```diff
@@ -3461,9 +3467,32 @@ function is_clone() {
                 $blog_id = $this->_storage->network_install_blog_id;
             }
 
+            if ( ! $this->_site->is_clone( $blog_id ) ) {
+                return false;
+            }
+
+            if ( ! $is_strict_mode ) {
+                return true;
+            }
+
             return (
-                trailingslashit( fs_strip_url_protocol( $this->_site->url ) ) !==
-                self::get_unfiltered_site_url( $blog_id, true, true )
+                ! empty( $this->_storage->has_clone_resolution_support ) &&
+                $this->_storage->has_clone_resolution_support
+            );
+        }
+
+        /**
+         * @author Leo Fajardo (@leorw)
+         * @since 2.5.1
+         */
+        function maybe_update_clone_resolution_support_flag() {
+            if ( isset( $this->_storage->has_clone_resolution_support ) ) {
+                return;
+            }
+
+            $this->_storage->has_clone_resolution_support = (

```

Comment:
@fajardoleo I think the name selected for the flag is misleading since the SDK does support clone resolution. I feel a better var name would be `hide_manual_resolution` (i. opposite meaning; ii. once this moved into the clone manager class it will use its storage, so the clone context is obvious and there's no need to include it in the name of the property), and instead of `$is_strict_mode` rename to `$only_if_manual_resolution_is_not_hidden`.

Reply:
@vovafeldman Is the `is_clone()` method supposed to be also moved to the clone manager class?

Reply:
@fajardoleo I think it will complicate things since the logic relies on `$this->_site`, which means that we'll need to load the site every time before calling it. So my intuition is to keep it in `Freemius`.